src/test/java/training/chessington/model/pieces/KingTest.java

```diff
@@ -1,7 +1,98 @@
 package training.chessington.model.pieces;
 
+import org.junit.Test;
+import training.chessington.model.Board;
+import training.chessington.model.Coordinates;
+import training.chessington.model.Move;
+import training.chessington.model.PlayerColour;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 import static training.chessington.model.pieces.PieceAssert.*;
 import static org.assertj.core.api.Assertions.*;
 
 public class KingTest {
+
+    @Test
+    public void kingCanMoveInAnyDirection() {
+        Board board = Board.empty();
+        King king = new King(PlayerColour.WHITE, board);
+        Coordinates kingPosition = new Coordinates(4, 4);
+        board.placePiece(kingPosition, king);
+        List<Move> validMoves = new ArrayList<>();
+
+        for (int row = 3; row <= 5; row++) {

```

Comment:
It's normally a good idea to make your tests as "stupid" as possible. Although this is a correct way of working out the valid destination squares, it is also exactly the code you've written in the actual `King` class, but with explicit values. So if there's a bug in it, this test won't pick it up.

Better to just list the expected destinations by hand.